File: training-data/ocaml/roman_numerals.ml
let values =
  [ (1000, "M"); (900, "CM"); (500, "D"); (400, "CD"); (100, "C"); (90, "XC");
    (50, "L"); (40, "XL"); (10, "X"); (9, "IX"); (5, "V"); (4, "IV"); (1, "I") ]

let rec to_roman n =
  if n = 0 then ""
  else
    let value, symbol = List.find (fun (v, _) -> v <= n) values in
    symbol ^ to_roman (n - value)

let () = print_endline (to_roman 1994)

File: training-data/prolog/roman_numerals.pl
roman_value(1000, 'M'). roman_value(900, 'CM'). roman_value(500, 'D').
roman_value(400, 'CD'). roman_value(100, 'C'). roman_value(90, 'XC').
roman_value(50, 'L'). roman_value(40, 'XL'). roman_value(10, 'X').
roman_value(9, 'IX'). roman_value(5, 'V'). roman_value(4, 'IV'). roman_value(1, 'I').

to_roman(0, '') :- !.
to_roman(N, Roman) :-
    roman_value(Value, Symbol),
    Value =< N, !,
    N1 is N - Value,
    to_roman(N1, Rest),
    atom_concat(Symbol, Rest, Roman).

:- to_roman(1994, Roman), writeln(Roman).
